### PR TITLE
Fixed decorations for single characters followed by each other

### PR DESCRIPTION
**Pull request by delijah**
#### What's the new behavior?

Looks like one if-statement was just done wrong before. Single character decorations followed by each other were not taken in consideration. Changing that if-statement (see changed files of this PR) fixes this edge case scenario. Existing tests and examples still work.

#### How does this change work?

This code part is so complicated, it's hard for me to describe what changed. Please have a look at the code and you will see.... Since i am not 100% sure how this function works, it would be great if someone that is more aware of the internals of this function, could have a look at this PR.

#### Have you checked that...?

- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #3693

### CODE REVIEW

File: packages/slate/test/interfaces/Text/decorations/in-a-row.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { Text } from 'slate'
+
+export const input = [
+  {
```
Comment (CameronAckermanSEL) [inline packages/slate/test/interfaces/Text/decorations/in-a-row.js:4]: This is a great test, but I'm wondering whether the name is correct. Are these two adjacent decorations or are they overlapping? 

```
0  1  2  3  4
a [b (c] d) e
```
Where decoration one is [] and decoration two is ().

Am I picturing this correctly, or does one of the ends of the decoration not actually contain the character at the index?

Comment (CameronAckermanSEL) [reply packages/slate/test/interfaces/Text/decorations/in-a-row.js:4]: Oh and a follow up, if this test is indeed "overlapping" decorations, then should we have one for adjacent decorations also?

Comment (delijah) [reply packages/slate/test/interfaces/Text/decorations/in-a-row.js:4]: Yeah sorry my english is not that good. This test is about adjacent decorations. I will change it.

Comment (delijah) [reply packages/slate/test/interfaces/Text/decorations/in-a-row.js]: Done :)

File: packages/slate/src/interfaces/text.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ export const Text = {
 
         // If the range starts after the leaf, or ends before it, continue.
         if (
-          start.offset > offset + length ||
+          start.offset >= offset + length ||
```
Comment (cudr) [inline packages/slate/src/interfaces/text.ts:114]: @delijah @CameronAckermanSEL, it looks like collapsed decorators will never be rendered at the end of the line and intersected after this change.

Comment (CameronAckermanSEL) [reply packages/slate/src/interfaces/text.ts:114]: Thanks, this is a good catch
